app/components/viewer/TimeLine.tsx

```diff
@@ -0,0 +1,166 @@
+import * as THREE from 'three';
+import {
+    Billboard, Sphere, Text, Plane,
+} from '@react-three/drei';
+import { useFrame } from '@react-three/fiber';
+import {
+    Slider, InputNumber, Row, Col, Button,
+} from 'antd';
+import { ReplayData } from '../../lib/api/apiRequests';
+import { getRaceTimeStr, timeDifference } from '../../lib/utils/time';
+
+export class TimeLineInfos {
+    currentRaceTime: number;
+    followedReplay: any;
+    constructor() {
+        this.currentRaceTime = 0;
+        this.followedReplay = null;
+    }
+}
+
+interface TimeLineViewProps {
+    replaysData: ReplayData[];
+    timeLineGlobal: TimeLineInfos;
+}
+
+// declare setInterval return variable outside to keep persistent reference for clearInterval after render
+let playInterval: any;
+
+const TimeLineView = ({ replaysData, timeLineGlobal }: TimeLineViewProps) => {
+    const [timeLineTime, setTimeLineTime] = useState<number>(0);
+    const [sampleInterval, setSampleInterval] = useState<number>(7);
+    const [playing, setPlaying] = useState<boolean>(false);
+
+    const min = 0;
+    let max = 0;
+
+    if (timeLineGlobal.followedReplay !== null) {
+        if (!replaysData.some((replay: ReplayData) => replay._id === timeLineGlobal.followedReplay._id)) {
+            timeLineGlobal.followedReplay = null;
+        }
+    }
+
+    if (replaysData.length === 0) {
+        if (timeLineTime !== 0) {
+            timeLineGlobal.currentRaceTime = 0;
+            setTimeLineTime(0);
+        }
+        if (playing) {
+            setPlaying(false);
+        }
+    }
+
+    timeLineGlobal.currentRaceTime = timeLineTime;
+
+    replaysData.forEach((replay) => {
+        if (replay.samples[replay.samples.length - 1].currentRaceTime > max) {
+            max = replay.samples[replay.samples.length - 1].currentRaceTime;
+        }
+    });
+
+    const onChange = (e: any) => {
+        setTimeLineTime(Math.round(e));
+        timeLineGlobal.currentRaceTime = Math.round(e);
+    };
+
+    const onClick = () => {
+        setPlaying(!playing);
+        if (playing) {
+            onChange(timeLineGlobal.currentRaceTime - 1);
+            clearInterval(playInterval);
+        } else {
+            playInterval = setInterval(() => {
+                if (timeLineGlobal.currentRaceTime + sampleInterval > max) {
+                    onChange(min);
+                } else {
+                    onChange(timeLineGlobal.currentRaceTime + sampleInterval);
+                }
+            }, 0);
+        }
+    };
+
+    const onChangeInterval = (e: number) => {
+        setSampleInterval(e);
+        if (playing) {
+            clearInterval(playInterval);
+            playInterval = setInterval(() => {
+                if (timeLineGlobal.currentRaceTime + sampleInterval > max) {
+                    onChange(min);
+                } else {
+                    onChange(timeLineGlobal.currentRaceTime + sampleInterval);
+                }
+            }, 0);
+        }
+    };
```

Comment:
These two methods are really similar, you might be able to extract parts of the method to avoid duplication